docs/usage/topology_aware_routing.md

```diff
@@ -0,0 +1,93 @@
+1. The Pods backing the Service should be spread on most of the availabel zones. This constraint should be ensured with appropriate scheduling constraints (Pod topology spread constraints, Pod (anti-)affinity). Enabling the feature for a Service with a single backing Pod or Pods all located in the same zone does not lead to a benefit.
+1. The component should be scaled up by `VerticalPodAutoscaler`. In case of an overload (a large portion of the of the traffic is originating from a given zone) the `VerticalPodAutoscaler` should provide better resource recommendations for the overloaded backing Pods.
+1. Consider the [`TopologyAwareHints` constraints](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/#constraints).
+
+> Note: The topology-aware routing feature is considered as alpha feature. Use it only for evaluation purposes.
+
+## Topology-aware Services in the Seed cluster
+
+##### etcd-main-client and etcd-events-client
+
+The `etcd-main-client` and `etcd-events-client` Services are topology-aware. They are consumed by the kube-apiserver.
+
+##### kube-apiserver
+
+The `kube-apiserver` Service is topology-aware. It is consumed by the controllers running in the Shoot control plane.
+
+##### gardener-resource-manager
+
+The `gardener-resource-manager` Service that is part of the Shoot control plane is topology-aware. The resource-manager serves webhooks and the Service is consumed by the kube-apiserver for the webhook communication.
+
+##### vpa-webhook
+
+The `vpa-webhook` Service that is part of the Shoot control plane is topology-aware. It is consumed by the kube-apiserver for the webhook communication.
+
+## How to enable the topology-aware routing?
+
+The topology-aware routing functionality can be enabled in the Seed specification:
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: Seed
+# ...
+spec:
+  settings:
+    topologyAwareRouting:
+      enabled: true
+```
+
+The topology-aware routing setting can be only enabled for a Seed cluster with more than one zone.
+gardenlet enables topology-aware Services only for Shoot control planes with failure tolerance type `zone` (`.spec.controlPlane.highAvailability.failureTolerance.type=zone`). Control plane Pods of non-HA Shoots and HA Shoots with failure tolerance type `node` are pinned to single zone. For more details, see [High Availability Of Deployed Components](../development/high-availability.md).
+
+⚠️ For K8s < 1.24 clusters, the topology-aware routing setting requires the Kubernetes `TopologyAwareHints` feature gate to be enabled for kube-apiserver, kube-controller-manager and kube-proxy. This is required because the `TopologyAwareHints` feature gate is disabled by default in K8s < 1.24. When `TopologyAwareHints` is disabled the kube-apiserver does not allow anything to be persisted in the `.endpoints[].hints` field in the EndpointSlice resource, the kube-controller-manager removes the hints and kube-proxy is not using topology-aware routing.
```

Comment:
```suggestion
⚠️ For K8s < 1.24 clusters, the topology-aware routing setting requires the Kubernetes `TopologyAwareHints` feature gate to be enabled for kube-apiserver, kube-controller-manager and kube-proxy. This is required because the `TopologyAwareHints` feature gate is disabled by default in K8s < 1.24. When `TopologyAwareHints` is disabled, the kube-apiserver does not allow anything to be persisted in the `.endpoints[].hints` field in the EndpointSlice resource. Also, the kube-controller-manager removes the hints, hence kube-proxy is not using topology-aware routing.
```